capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,288 @@
+            resultMap.put("count", results.getProperty("count"));
+        }
+
+        resultMap.put("year", year);
+        resultMap.put("week", week);
+
+        return resultMap;
+    }
+
+    /**
+     * Gets the weekly pageViews for the entire year of a specific restaurant.
+     * @param year we are looking for
+     * @param restaurantKey of restaurant we want the data.
+     * @return list of maps in the following way: List<Map<String, Object>>
+     *          [
+     *                 {
+     *                      "year": <String, the year number>,
+     *                      "week": <String, this week's number in relation to the year.>,
+     *                      "count": <String, amount of page views the restaurant had.>
+     *                  },
+     *                  ...,
+     *          ]
+     */
+    public List<Map<String, Object>> getYearRestaurantPageViews(int year, String restaurantKey) {
+
+        CompositeFilter compositeFilter = new CompositeFilter(Query.CompositeFilterOperator.AND,
+                Arrays.asList(
+                        new FilterPredicate("restaurantKey", Query.FilterOperator.EQUAL, restaurantKey),
+                        new FilterPredicate("year", Query.FilterOperator.EQUAL, year)
+                ));
+        Query query = new Query("PageViews").setFilter(compositeFilter).addSort("week", Query.SortDirection.ASCENDING);
+        PreparedQuery preparedQuery = dataStore.prepare(query);
+        List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withDefaults());
+
+        List<Map<String, Object>> resultList = new ArrayList<>();
+
+        Map<String, Object> map;
+        for (Entity entity : results) {
+            map = new HashMap<>();
+            map.put("week", entity.getProperty("week").toString());
+            map.put("count", Integer.parseInt(entity.getProperty("count").toString()));
+            resultList.add(map);
+        }
+
+        return resultList;
+    }
+
+    /**
+     * Gets all pageViews for the entire system.
+     * @return returns a Map<String, Object> in the following format:
+     *          {
+     *              "firstDate":
+     *                  {
+     *                      "week": <int>
+     *                      "year": <int>
+     *                  },
+     *              "clickData":
+     *                  [
+     *                      {
+     *                          "restaurantName": <String>
+     *                          "week": <int>
+     *                          "year": <int>
+     *                          "numClicks": <int>
+     *                      },
+     *                      ...
+     *                  ]
+     *          }
+     *
+     *          "clickData" represents all the click data that we have.
+     *          "firstDate" represent the first time data was collected in our system.
+     */
+    public Map<String, Object> getAllPageViews() {
```

Comment:
Same comment as before about the structured type. :p

Reply:
Fixed as well!